src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -454,25 +468,48 @@ public AgentInfo invoke(File nodeRoot, VirtualChannel virtualChannel) throws IOE
                 os = OsType.WINDOWS;
             } else if(Platform.current() == Platform.UNIX && System.getProperty("os.name").equals("z/OS")) {
                 os = OsType.ZOS;
+            } else if(Platform.current() == Platform.UNIX && System.getProperty("os.name").equals("FreeBSD")) {
+                os = OsType.FREEBSD;
```

Comment:
Not that we need to fix it here, but do you know what's wrong with the Go binary on FreeBSD? Is it a Go problem, or do we need to cross-compile for it separately or something?

Reply:
So I just confirmed, i forgot i could have go cross-compile for FreeBSD.  
The OS does have a linux compatability mode, but yeah, it does have some caveats. Probably best to do the freebsd cross-compile if we ever do want to do Freebsd compatibility.